src/components/icons/icons.tsx

```diff
@@ -22,18 +26,101 @@ export const IonIcon = ({
   type,
   color = defaultColor,
   size = defaultSize,
+  highlight = 'none',
 }: IonIconProps) => {
+  const defaultContainerStyle = {
+    color: 'transparent',
+    size: 'unset',
+  };
+
+  const getCircleProportion = (
```

Comment:
You could simplify these functions if you treat what is a const outside the component, and leave it inside only the things that are conditional to prop values. But, it's up to you to decide, it's fine the way it is right now tbh.